{
  "uid" : "e8735087e1df1199",
  "name" : "Check the display of the add to wish and add to compare buttons",
  "fullName" : "tests.test_main_page.TestHotSellerSection#test_tc_06_01_15_check_the_display_of_the_card_buttons",
  "historyId" : "6e2a298dcd92bfc3e66f7769ef7fb2ac",
  "time" : {
    "start" : 1737855772294,
    "stop" : 1737855777638,
    "duration" : 5344
  },
  "description" : "\n            This test will check that the add to wishlist and add to compare buttons are visible on the screen\n            ",
  "descriptionHtml" : "<pre><code>        This test will check that the add to wishlist and add to compare buttons are visible on the screen\n</code></pre>\n",
  "status" : "skipped",
  "statusMessage" : "XFAIL In CI the test failed in PR#182\n\nselenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55ad2590a7ca <unknown>\n#1 0x55ad254022f0 <unknown>\n#2 0x55ad25452035 <unknown>\n#3 0x55ad25452251 <unknown>\n#4 0x55ad25498054 <unknown>\n#5 0x55ad254769dd <unknown>\n#6 0x55ad254953b3 <unknown>\n#7 0x55ad25476753 <unknown>\n#8 0x55ad25443baa <unknown>\n#9 0x55ad25444dfe <unknown>\n#10 0x55ad258d538b <unknown>\n#11 0x55ad258d9307 <unknown>\n#12 0x55ad258c1e7c <unknown>\n#13 0x55ad258d9ec7 <unknown>\n#14 0x55ad258a624f <unknown>\n#15 0x55ad258f92f8 <unknown>\n#16 0x55ad258f94c0 <unknown>\n#17 0x55ad25909646 <unknown>\n#18 0x7fe46fc9ca94 <unknown>\n#19 0x7fe46fd29c3c <unknown>",
  "statusTrace" : "self = <test_main_page.TestMainPage.TestHotSellerSection object at 0x7fc2e82e9c50>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"16e69224a7885f9b2bbce91030f75925\")>\nitem = 'add_to_card'\n\n    @pytest.mark.xfail(reason=\"In CI the test failed in PR#182\")\n    @allure.title(\"Check the display of the add to wish and add to compare buttons\")\n    @pytest.mark.parametrize(\"item\", product_card_button)\n    def test_tc_06_01_15_check_the_display_of_the_card_buttons(self, driver, item):\n        \"\"\"\n        This test will check that the add to wishlist and add to compare buttons are visible on the screen\n        \"\"\"\n        page = MainPage(driver, MAIN_PAGE_URL)\n        page.open()\n>       check_display = page.check_element_display(item)\n\ntests/test_main_page.py:96: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/main_page.py:122: in check_element_display\n    product_card = self.element_is_visible(self.locators.PRODUCT_CARD)\npages/base_page.py:29: in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\npages/base_page.py:50: in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"16e69224a7885f9b2bbce91030f75925\")>\nmethod = <function presence_of_element_located.<locals>._predicate at 0x7fc2e7bbfa60>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x55ad2590a7ca <unknown>\nE       #1 0x55ad254022f0 <unknown>\nE       #2 0x55ad25452035 <unknown>\nE       #3 0x55ad25452251 <unknown>\nE       #4 0x55ad25498054 <unknown>\nE       #5 0x55ad254769dd <unknown>\nE       #6 0x55ad254953b3 <unknown>\nE       #7 0x55ad25476753 <unknown>\nE       #8 0x55ad25443baa <unknown>\nE       #9 0x55ad25444dfe <unknown>\nE       #10 0x55ad258d538b <unknown>\nE       #11 0x55ad258d9307 <unknown>\nE       #12 0x55ad258c1e7c <unknown>\nE       #13 0x55ad258d9ec7 <unknown>\nE       #14 0x55ad258a624f <unknown>\nE       #15 0x55ad258f92f8 <unknown>\nE       #16 0x55ad258f94c0 <unknown>\nE       #17 0x55ad25909646 <unknown>\nE       #18 0x7fe46fc9ca94 <unknown>\nE       #19 0x7fe46fd29c3c <unknown>\n\n/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "clear_allure_result_folder",
    "time" : {
      "start" : 1737854757354,
      "stop" : 1737854757354,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "driver",
    "time" : {
      "start" : 1737855771061,
      "stop" : 1737855772294,
      "duration" : 1233
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "_session_faker",
    "time" : {
      "start" : 1737854757290,
      "stop" : 1737854757353,
      "duration" : 63
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "testStage" : {
    "description" : "\n            This test will check that the add to wishlist and add to compare buttons are visible on the screen\n            ",
    "status" : "skipped",
    "statusMessage" : "XFAIL In CI the test failed in PR#182\n\nselenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55ad2590a7ca <unknown>\n#1 0x55ad254022f0 <unknown>\n#2 0x55ad25452035 <unknown>\n#3 0x55ad25452251 <unknown>\n#4 0x55ad25498054 <unknown>\n#5 0x55ad254769dd <unknown>\n#6 0x55ad254953b3 <unknown>\n#7 0x55ad25476753 <unknown>\n#8 0x55ad25443baa <unknown>\n#9 0x55ad25444dfe <unknown>\n#10 0x55ad258d538b <unknown>\n#11 0x55ad258d9307 <unknown>\n#12 0x55ad258c1e7c <unknown>\n#13 0x55ad258d9ec7 <unknown>\n#14 0x55ad258a624f <unknown>\n#15 0x55ad258f92f8 <unknown>\n#16 0x55ad258f94c0 <unknown>\n#17 0x55ad25909646 <unknown>\n#18 0x7fe46fc9ca94 <unknown>\n#19 0x7fe46fd29c3c <unknown>",
    "statusTrace" : "self = <test_main_page.TestMainPage.TestHotSellerSection object at 0x7fc2e82e9c50>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"16e69224a7885f9b2bbce91030f75925\")>\nitem = 'add_to_card'\n\n    @pytest.mark.xfail(reason=\"In CI the test failed in PR#182\")\n    @allure.title(\"Check the display of the add to wish and add to compare buttons\")\n    @pytest.mark.parametrize(\"item\", product_card_button)\n    def test_tc_06_01_15_check_the_display_of_the_card_buttons(self, driver, item):\n        \"\"\"\n        This test will check that the add to wishlist and add to compare buttons are visible on the screen\n        \"\"\"\n        page = MainPage(driver, MAIN_PAGE_URL)\n        page.open()\n>       check_display = page.check_element_display(item)\n\ntests/test_main_page.py:96: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/main_page.py:122: in check_element_display\n    product_card = self.element_is_visible(self.locators.PRODUCT_CARD)\npages/base_page.py:29: in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\npages/base_page.py:50: in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"16e69224a7885f9b2bbce91030f75925\")>\nmethod = <function presence_of_element_located.<locals>._predicate at 0x7fc2e7bbfa60>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x55ad2590a7ca <unknown>\nE       #1 0x55ad254022f0 <unknown>\nE       #2 0x55ad25452035 <unknown>\nE       #3 0x55ad25452251 <unknown>\nE       #4 0x55ad25498054 <unknown>\nE       #5 0x55ad254769dd <unknown>\nE       #6 0x55ad254953b3 <unknown>\nE       #7 0x55ad25476753 <unknown>\nE       #8 0x55ad25443baa <unknown>\nE       #9 0x55ad25444dfe <unknown>\nE       #10 0x55ad258d538b <unknown>\nE       #11 0x55ad258d9307 <unknown>\nE       #12 0x55ad258c1e7c <unknown>\nE       #13 0x55ad258d9ec7 <unknown>\nE       #14 0x55ad258a624f <unknown>\nE       #15 0x55ad258f92f8 <unknown>\nE       #16 0x55ad258f94c0 <unknown>\nE       #17 0x55ad25909646 <unknown>\nE       #18 0x7fe46fc9ca94 <unknown>\nE       #19 0x7fe46fd29c3c <unknown>\n\n/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException",
    "steps" : [ {
      "name" : "Open a browser",
      "time" : {
        "start" : 1737855772294,
        "stop" : 1737855772522,
        "duration" : 228
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "Checking the display of an element on the screen",
      "time" : {
        "start" : 1737855772522,
        "stop" : 1737855777638,
        "duration" : 5116
      },
      "status" : "broken",
      "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55ad2590a7ca <unknown>\n#1 0x55ad254022f0 <unknown>\n#2 0x55ad25452035 <unknown>\n#3 0x55ad25452251 <unknown>\n#4 0x55ad25498054 <unknown>\n#5 0x55ad254769dd <unknown>\n#6 0x55ad254953b3 <unknown>\n#7 0x55ad25476753 <unknown>\n#8 0x55ad25443baa <unknown>\n#9 0x55ad25444dfe <unknown>\n#10 0x55ad258d538b <unknown>\n#11 0x55ad258d9307 <unknown>\n#12 0x55ad258c1e7c <unknown>\n#13 0x55ad258d9ec7 <unknown>\n#14 0x55ad258a624f <unknown>\n#15 0x55ad258f92f8 <unknown>\n#16 0x55ad258f94c0 <unknown>\n#17 0x55ad25909646 <unknown>\n#18 0x7fe46fc9ca94 <unknown>\n#19 0x7fe46fd29c3c <unknown>\n\n",
      "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/main_page.py\", line 122, in check_element_display\n    product_card = self.element_is_visible(self.locators.PRODUCT_CARD)\n                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 29, in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n",
      "steps" : [ {
        "name" : "Find a visible element",
        "time" : {
          "start" : 1737855772522,
          "stop" : 1737855777637,
          "duration" : 5115
        },
        "status" : "broken",
        "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55ad2590a7ca <unknown>\n#1 0x55ad254022f0 <unknown>\n#2 0x55ad25452035 <unknown>\n#3 0x55ad25452251 <unknown>\n#4 0x55ad25498054 <unknown>\n#5 0x55ad254769dd <unknown>\n#6 0x55ad254953b3 <unknown>\n#7 0x55ad25476753 <unknown>\n#8 0x55ad25443baa <unknown>\n#9 0x55ad25444dfe <unknown>\n#10 0x55ad258d538b <unknown>\n#11 0x55ad258d9307 <unknown>\n#12 0x55ad258c1e7c <unknown>\n#13 0x55ad258d9ec7 <unknown>\n#14 0x55ad258a624f <unknown>\n#15 0x55ad258f92f8 <unknown>\n#16 0x55ad258f94c0 <unknown>\n#17 0x55ad25909646 <unknown>\n#18 0x7fe46fc9ca94 <unknown>\n#19 0x7fe46fd29c3c <unknown>\n\n",
        "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 29, in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n",
        "steps" : [ {
          "name" : "Find a present element",
          "time" : {
            "start" : 1737855772522,
            "stop" : 1737855777637,
            "duration" : 5115
          },
          "status" : "broken",
          "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55ad2590a7ca <unknown>\n#1 0x55ad254022f0 <unknown>\n#2 0x55ad25452035 <unknown>\n#3 0x55ad25452251 <unknown>\n#4 0x55ad25498054 <unknown>\n#5 0x55ad254769dd <unknown>\n#6 0x55ad254953b3 <unknown>\n#7 0x55ad25476753 <unknown>\n#8 0x55ad25443baa <unknown>\n#9 0x55ad25444dfe <unknown>\n#10 0x55ad258d538b <unknown>\n#11 0x55ad258d9307 <unknown>\n#12 0x55ad258c1e7c <unknown>\n#13 0x55ad258d9ec7 <unknown>\n#14 0x55ad258a624f <unknown>\n#15 0x55ad258f92f8 <unknown>\n#16 0x55ad258f94c0 <unknown>\n#17 0x55ad25909646 <unknown>\n#18 0x7fe46fc9ca94 <unknown>\n#19 0x7fe46fd29c3c <unknown>\n\n",
          "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n",
          "steps" : [ ],
          "attachments" : [ ],
          "parameters" : [ {
            "name" : "locator",
            "value" : "('css selector', \"li[class='product-item']:nth-child(4) div[class='product-item-info']\")"
          }, {
            "name" : "timeout",
            "value" : "5"
          } ],
          "stepsCount" : 0,
          "attachmentsCount" : 0,
          "shouldDisplayMessage" : true,
          "hasContent" : true
        } ],
        "attachments" : [ ],
        "parameters" : [ {
          "name" : "locator",
          "value" : "('css selector', \"li[class='product-item']:nth-child(4) div[class='product-item-info']\")"
        }, {
          "name" : "timeout",
          "value" : "10"
        } ],
        "stepsCount" : 1,
        "attachmentsCount" : 0,
        "shouldDisplayMessage" : false,
        "hasContent" : true
      } ],
      "attachments" : [ ],
      "parameters" : [ {
        "name" : "item",
        "value" : "'add_to_card'"
      } ],
      "stepsCount" : 2,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    } ],
    "attachments" : [ {
      "uid" : "6cf29f697451f8e2",
      "name" : "Screenshot",
      "source" : "6cf29f697451f8e2.png",
      "type" : "image/png",
      "size" : 25650
    }, {
      "uid" : "d255f9bbad352162",
      "name" : "HTML source",
      "source" : "d255f9bbad352162.html",
      "type" : "text/html",
      "size" : 258745
    } ],
    "parameters" : [ ],
    "stepsCount" : 4,
    "attachmentsCount" : 2,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ {
    "name" : "clear_allure_result_folder::0",
    "time" : {
      "start" : 1737857182853,
      "stop" : 1737857182853,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "driver::0",
    "time" : {
      "start" : 1737855777757,
      "stop" : 1737855777820,
      "duration" : 63
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "epic",
    "value" : "Main Page"
  }, {
    "name" : "feature",
    "value" : "Testing Hot Seller Section"
  }, {
    "name" : "tag",
    "value" : "@pytest.mark.xfail(reason='In CI the test failed in PR#182')"
  }, {
    "name" : "parentSuite",
    "value" : "tests"
  }, {
    "name" : "suite",
    "value" : "test_main_page"
  }, {
    "name" : "subSuite",
    "value" : "TestMainPage"
  }, {
    "name" : "host",
    "value" : "fv-az1922-652"
  }, {
    "name" : "thread",
    "value" : "2797-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.test_main_page"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ {
    "name" : "item",
    "value" : "'add_to_card'"
  } ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ ],
    "tags" : [ "@pytest.mark.xfail(reason='In CI the test failed in PR#182')" ]
  },
  "source" : "e8735087e1df1199.json",
  "parameterValues" : [ "'add_to_card'" ]
}